tests/integration/test_hyper_transformer.py

```diff
@@ -101,110 +192,193 @@ def get_transformers():
     }
 
 
-def test_hypertransformer_with_transformers():
-    data = get_input_data_without_nan()
-    transformers = get_transformers()
-
-    ht = HyperTransformer(transformers)
-    ht.fit(data)
-    transformed = ht.transform(data)
-
-    expected = get_transformed_data()
+@patch('rdt.transformers.get_default_transformers')
+def test_hypertransformer_no_inputs(default_transformers_mock):
+    """Test the HyperTransformer with no transformers provided.
 
-    np.testing.assert_allclose(
-        transformed.sort_index(axis=1).values,
-        expected.sort_index(axis=1).values
-    )
-
-    reversed_data = ht.reverse_transform(transformed)
-
-    original_names = data.pop('names')
-    reversed_names = reversed_data.pop('names')
+    This tests that if no parameters are provided to the HyperTransformer,
+    the ``default_transformers`` method will be used to determine which
+    transformers to use for each field. Any output of a transformer that
+    is not ML ready should be recursively transformed till it is.
 
-    pd.testing.assert_frame_equal(data.sort_index(axis=1), reversed_data.sort_index(axis=1))
+    Setup:
+        - A dummy transformer that just returns the input data.
+        - Another dummy transformer that returns data that is not ML ready.
+        - A dummy dict that replaces the ``default_transformers`` method
+        so that all the field types use the dummy transformers created.
 
-    for name in original_names:
-        assert name not in reversed_names
+    Input:
+        - A dataframe with every data type.
 
+    Expected behavior:
+        - The transformed data should contain all the ML ready data.
+        - The reverse transformed data should be the same as the input.
+    """
+    get_default_transformer.cache_clear()
+    default_transformers_mock.return_value = {
+        'numerical': DummyTransformer,
+        'integer': DummyTransformer,
+        'float': DummyTransformer,
+        'categorical': DummyTransformer,
+        'boolean': DummyTransformer,
+        'datetime': DummyTransformerNotMLReady,
+    }
+    data = get_input_data_without_nan()
+    expected_transformed = data.drop('datetime', axis=1)
+    expected_transformed.columns = [
+        'integer.value',
+        'float.value',
+        'categorical.value',
+        'bool.value',
+        'names.value'
+    ]
+    expected_transformed['datetime.value.value'] = [
+        '2010-02-01', '2010-01-01', '2010-02-01', '2010-01-01'
+    ]
+    expected_reversed = data.copy()
 
-def test_hypertransformer_with_transformers_nan_data():
-    data = get_input_data_with_nan()
-    transformers = get_transformers()
-
-    ht = HyperTransformer(transformers)
+    ht = HyperTransformer()
     ht.fit(data)
     transformed = ht.transform(data)
 
-    expected = get_transformed_nan_data()
+    pd.testing.assert_frame_equal(
+        transformed.sort_index(axis=1),
```

Comment:
I have doubts about this `sort_index`. Should the output data not be already sorted like the input by default?

Reply:
Good point. I didn't actually sort in the `reverse_transform` method but I suppose we could.

Reply:
This was because the way I made the expected output was incorrect. I can change that